src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -519,13 +529,33 @@ Task StartNegotiation()
                                         }
                                         if (!string.IsNullOrEmpty(negotiationResponse.RedirectUrl))
                                         {
-                                            if (!negotiationResponse.RedirectUrl.EndsWith("/"))
+                                            try
                                             {
-                                                negotiationResponse.RedirectUrl += "/";
+                                                var redirectUrl = new UriBuilder(negotiationResponse.RedirectUrl);
```

Comment:
Do you think we should just split the string on '?' like we do for the JS client instead of attempting parsing it with UriBuilder?

I haven't come up with a case where it would work differently given a RedirectUrl that's valid enough to attempt a request with.

Reply:
This in most cases will only be happening once so creating the UriBuilder shouldn't have a big impact.